File: src/commands/add.ts
import inquirer from 'inquirer';
import chalk from 'chalk';
import { ConfigManager, Project } from '../config';

export async function addCommand(name?: string): Promise<void> {
  const configManager = new ConfigManager();

  try {
    let projectName = name;

    if (!projectName) {
      const { inputName } = await inquirer.prompt([
        {
          type: 'input',
          name: 'inputName',
          message: 'Enter project name:',
          validate: (input: string) => {
            if (!input.trim()) {
              return 'Project name cannot be empty';
            }
            if (configManager.projectExists(input.trim())) {
              return `Project "${input.trim()}" already exists`;
            }
            return true;
          }
        }
      ]);
      projectName = inputName.trim();
    } else {
      projectName = projectName.trim();
      if (configManager.projectExists(projectName)) {
        console.error(chalk.red(`Project "${projectName}" already exists`));
        return;
      }
    }

    const project: Project = {
      name: projectName!
    };

    configManager.addProject(project);
    console.log(chalk.green(`Project "${projectName}" added successfully!`));

    const projects = configManager.getProjects();
    if (projects.length === 1) {
      console.log(chalk.blue(`"${projectName}" is now the current project.`));
    }
  } catch (error: any) {
    if (error.isTtyError) {
      console.error(chalk.red('Prompt couldn\'t be rendered in the current environment'));
    } else if (error instanceof Error) {
      console.error(chalk.red('An error occurred:'), error.message);
    } else {
      console.error(chalk.red('An unknown error occurred'));
    }
  }
}

File: src/commands/add.rs
use crate::config::{ConfigManager, Project};
use anyhow::Result;
use colored::*;
use inquire::Text;

pub fn execute(name: Option<String>) -> Result<()> {
    let mut config_manager = ConfigManager::new()?;

    let project_name = if let Some(name) = name {
        let trimmed_name = name.trim().to_string();
        if config_manager.project_exists(&trimmed_name) {
            anyhow::bail!("Project '{}' already exists", trimmed_name);
        }
        trimmed_name
    } else {
        Text::new("Enter project name:")
            .with_validator(|input: &str| {
                let trimmed = input.trim();
                if trimmed.is_empty() {
                    Ok(inquire::validator::Validation::Invalid(
                        "Project name cannot be empty".into(),
                    ))
                } else if ConfigManager::new()
                    .map(|cm| cm.project_exists(trimmed))
                    .unwrap_or(false)
                {
                    Ok(inquire::validator::Validation::Invalid(
                        format!("Project '{}' already exists", trimmed).into(),
                    ))
                } else {
                    Ok(inquire::validator::Validation::Valid)
                }
            })
            .prompt()?
            .trim()
            .to_string()
    };

    let project = Project {
        name: project_name.clone(),
        path: None,
        description: None,
        browser: None,
        commands: None,
    };

    config_manager.add_project(project)?;
    println!("{}", format!("Project '{}' added successfully!", project_name).green());

    let projects = config_manager.get_projects();
    if projects.len() == 1 {
        println!("{}", format!("'{}' is now the current project.", project_name).blue());
    }

    Ok(())
}